src/Disks/S3/DiskS3.cpp

```diff
@@ -632,6 +645,12 @@ std::unique_ptr<ReadBufferFromFileBase> DiskS3::readFile(const String & path, si
 std::unique_ptr<WriteBufferFromFileBase> DiskS3::writeFile(const String & path, size_t buf_size, WriteMode mode, size_t estimated_size, size_t)
```

Comment:
Also `truncateFile`.

Reply:
`truncateFile` is not implemented for DiskS3 yet.

Reply:
Ok. It means Log, TinyLog, StripeLog cannot work on S3 in some cases.